src/main/java/com/notably/model/suggestion/SuggestionModelImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.notably.model.suggestion;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.property.Property;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+/**
+ * The implementation class of SuggestionModel.
+ */
+public class SuggestionModelImpl implements SuggestionModel {
```

Comment:
Right now when SuggestionModelImpl is instantiated, it uses default constructor right? This would mean accessing the attributes might result in NPE ?

Reply:
Good catch @HemanshuGandhi !

Yes, @johannagwan , you should initialize all the fields of this model to an "empty" version.